### ISSUE TITLE
Type Annotation are not allowed for enum members when using nonmember

### ISSUE THREAD

**Issue by kurt-rhee** [microsoft/pyright#10941]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

If you have a question about typing or a behavior that you’re seeing in Pyright (as opposed to a bug report or enhancement request), consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Describe the bug**
`Type annotations are not allowed for enum members` when trying to use a `nonmember()`

I tried to follow the example here, but I am new to using nonmember, so I could be doing something wrong.
https://github.com/microsoft/pyright/issues/7265

**Code or Screenshots**
If possible, provide a minimal, self-contained code sample (surrounded by triple back ticks) to demonstrate the issue. The code should define or import all referenced symbols.

```python
from enum import IntEnum, nonmember

class BaseIntEnum(IntEnum):
    _db_table: str | None = nonmember(None)
    _db_id_column = nonmember(None)
    _db_name_column = nonmember(None)
```

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension, a language server in another editor, integrated into Pylance, or the command-line tool? Which version?

Using in Zed editor, but to test I used pyright-play.net
